include/LcdFloatSpinBox.h

```diff
@@ -0,0 +1,97 @@
+/*
+ * LcdFloatSpinBox.h - class LcdFloatSpinBox (LcdSpinBox for floats)
+ *
+ * Copyright (c) 2005-2014 Tobias Doerffel <tobydox/at/users.sourceforge.net>
+ * Copyright (c) 2020 Martin Pavelek <he29.HS/at/gmail.com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+
+#ifndef LCD_FLOATSPINBOX_H
+#define LCD_FLOATSPINBOX_H
+
+#include <QString>
+
+#include "LcdWidget.h"
+#include "AutomatableModelView.h"
+
+
+class LMMS_EXPORT LcdFloatSpinBox : public QWidget, public FloatModelView
+{
+	Q_OBJECT
+public:
+	LcdFloatSpinBox(int numWhole, int numFrac, QWidget* parent, const QString& name = QString());
+	LcdFloatSpinBox(int numWhole, int numFrac, const QString& style, QWidget* parent, const QString& name = QString());
```

Comment:
Seems like a good idea to me. I was first a little worried about making the order of parameters inconsistent with `LcdSpinBox` (to avoid possible mistakes when converting `LcdSpinBox` to `LcdFloatSpinBox`), but since there is already one extra parameter and it is not something that is going to be done often, I think it makes sense to do the change.

Reply:
Hm, never mind, it probably wouldn't work well with the overloads. If changed to:
```cpp
LcdFloatSpinBox(int numWhole, int numFrac, const QString& name = QString(), QWidget* parent = nullptr);
LcdFloatSpinBox(int numWhole, int numFrac, const QString& style, const QString& name = QString(), QWidget* parent = nullptr);
```
there would be no way to tell whether `LcdFloatSpinBox(2, 3, "abcd")` is a call to create an instance with the name "abcd" and default parent, or an instance with style = "abcd" and default name and parent.

`grep`ing through the code, I don't actually see any call to `LcdSpinBox()` that would use the style definition, and `initUI()` in `LcdWidget` looks like the implementation is kind of half unfinished:
```cpp
 // We should make a factory for these or something.
 //m_lcdPixmap = new QPixmap( embed::getIconPixmap( QString( "lcd_" + style ).toUtf8().constData() ) );
 //m_lcdPixmap = new QPixmap( embed::getIconPixmap( "lcd_19green" ) ); // TODO!!
 m_lcdPixmap = new QPixmap( embed::getIconPixmap( QString( "lcd_" + style ).toUtf8().constData() ) );
```
(I'm not quite sure what the poet meant to say by these comments..)

So the question is whether we should even bother keeping the constructor with custom style support. I would hope that LMMS eventually moves away from bitmaps and the LCD will be drawn using a style defined in the CSS, so it would not be needed anyway.

Reply:
One possible solution to resolve ambiguity:
```cpp
LcdFloatSpinBox(int numWhole, int numFrac, const QString& name = QString(), QWidget* parent = nullptr);
LcdFloatSpinBox(int numWhole, int numFrac, const QString& style, const QString& name, QWidget* parent = nullptr);
```

Reply:
I guess this always tends to happen when you have a lot of default arguments. But I don't think the current approach of using the position of the parent argument to disambiguate these is particularly intuitive. I like PhysSong's solution for now, but it does look like we're moving away from pixmaps over time.

Reply:
Changed to the format suggested by PhysSong.